File: lesson_1/src/c_examples/06_functions.c
/*-------------------------------
  File: functions_example.c
-------------------------------*/
#include <stdio.h>

/* Function prototype */
int square(int x);

int main() {
    int n = 7;
    int result;

    /* Call the function */
    result = square(n);

    printf("The square of %d is %d\n", n, result);

    return 0;
}

/* Function to compute square */
int square(int x) {
    return x * x;
}

File: lesson_1/src/fortran_examples/06_functions.f90
!-------------------------------
! File: functions_example.f90
!-------------------------------
PROGRAM test_functions
INTEGER :: n, result

    ! Input value
    n = 7
    ! Call the function
    result = square(n)

    PRINT *, 'The square of', n, 'is', result
END PROGRAM test_functions

! Function to compute square
INTEGER FUNCTION square(x)
INTEGER, INTENT(IN) :: x

    square = x * x
END FUNCTION square
